pkg/build/generator/generator.go

```diff
@@ -258,24 +288,57 @@ func (g *BuildGenerator) resolveImageRepoReference(ctx kapi.Context, from *kapi.
 	} else {
 		namespace = defaultNamespace
 	}
-	imageStream, err := g.Client.GetImageStream(kapi.WithNamespace(ctx, namespace), from.Name)
-	if err != nil {
-		if errors.IsNotFound(err) {
+
+	switch from.Kind {
+	case "ImageStreamImage":
+		bits := strings.Split(from.Name, "@")
+		stream, err := g.Client.GetImageStream(kapi.WithNamespace(ctx, namespace), bits[0])
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return "", err
+			}
+			return "", fatalError{err}
+		}
+		ref, err := imageapi.DockerImageReferenceForStream(stream)
+		if err != nil {
 			return "", err
 		}
-		return "", fatalError{err}
-	}
-
-	if imageStream == nil || len(imageStream.Status.DockerImageRepository) == 0 {
-		return "", fmt.Errorf("could not resolve image stream %s/%s", namespace, from.Name)
-	}
-	glog.V(4).Infof("Found image stream %s/%s", namespace, imageStream.Name)
+		ref.ID = bits[1]
+		return ref.String(), nil
+	case "ImageStreamTag":
+		bits := strings.Split(from.Name, ":")
+		stream, err := g.Client.GetImageStream(kapi.WithNamespace(ctx, namespace), bits[0])
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return "", err
+			}
+			return "", fatalError{err}
```

Comment:
actually I misspoke on what this is doing.  this is wholly within the generate.go logic (this method is private) and used to determine if there's a fundamental issue with an ImageRepo reference (in which case the error is returned to the component that requested the build and the build is aborted) or if it's a manageable error in which case we continue with creating the build, in this case knowing that the ImageRepo is not yet there, but presumably will be in the future.